workspaces/linguist/plugins/catalog-backend-module-linguist-tags-processor/src/processor/LinguistTagsProcessor.ts

```diff
@@ -0,0 +1,294 @@
+ *
+ * @public
+ * */
+export class LinguistTagsProcessor implements CatalogProcessor {
+  private logger: LoggerService;
+  private discovery: DiscoveryService;
+  private auth: AuthService;
+  private loggerMeta = { plugin: 'LinguistTagsProcessor' };
+  private languageMap: Record<string, string | undefined> = {};
+  private tagPrefix: string = '';
+  private shouldProcessEntity: ShouldProcessEntity = (entity: Entity) => {
+    return entity.kind === 'Component';
+  };
+  private cacheTTLMilliseconds: number;
+  private bytesThreshold = 0;
+  private languageTypes: LanguageType[] = ['programming'];
+
+  getProcessorName(): string {
+    return 'LinguistTagsProcessor';
+  }
+
+  constructor(options: LinguistTagsProcessorOptions) {
+    this.logger = options.logger;
+    this.discovery = options.discovery;
+    this.auth = options.auth;
+    if (options.shouldProcessEntity) {
+      this.shouldProcessEntity = options.shouldProcessEntity;
+    }
+    this.cacheTTLMilliseconds = durationToMilliseconds(
+      options.cacheTTL || { minutes: 30 },
+    );
+    if (options.bytesThreshold) {
+      this.bytesThreshold = options.bytesThreshold;
+    }
+    if (options.languageTypes) {
+      this.languageTypes = options.languageTypes;
+    }
+    if (options.languageMap) {
+      this.languageMap = options.languageMap;
+    }
+    if (options.tagPrefix) {
+      this.tagPrefix = options.tagPrefix;
+    }
+  }
+
+  static fromConfig(
+    config: Config,
+    options: LinguistTagsProcessorOptions,
+  ): LinguistTagsProcessor {
+    const c = config.getOptionalConfig('linguist.tagsProcessor');
+    if (c) {
+      options.bytesThreshold ??= c.getOptionalNumber('bytesThreshold');
+      options.languageTypes ??= c.getOptionalStringArray(
+        'languageTypes',
+      ) as LanguageType[];
+      options.languageMap ??= c.getOptional('languageMap');
+      options.cacheTTL ??= c.getOptional('cacheTTL');
+      options.tagPrefix ??= c.getOptional('tagPrefix');
+    }
+
+    return new LinguistTagsProcessor(options);
+  }
+
+  /**
+   * Given an entity ref, fetches the language breakdown from the Linguist backend HTTP API.
+   * @param entityRef - stringified entity ref
+   * @returns The language breakdown
+   */
+  private async getLanguagesFromLinguistAPI(
+    entityRef: string,
+  ): Promise<string[]> {
+    this.logger.debug(`Fetching languages from linguist API`, {
+      ...this.loggerMeta,
+      entityRef,
+    });
+    const { token } = await this.auth.getPluginRequestToken({
+      onBehalfOf: await this.auth.getOwnServiceCredentials(),
+      targetPluginId: 'linguist',
+    });
+    const baseUrl = await this.discovery.getBaseUrl('linguist');
+    const linguistApi = new URL(`${baseUrl}/entity-languages`);
+    linguistApi.searchParams.append('entityRef', entityRef);
+    const linguistData = await fetch(linguistApi, {
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: `Bearer ${token}`,
```

Comment:
I am now finally getting to the point of upgrading my backstage instance to the new backend, and this line seems to be the key to making this work with Auth, so thanks for this <3

Reply:
You're welcome!